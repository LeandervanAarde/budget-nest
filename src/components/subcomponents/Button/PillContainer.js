import React from 'react';
import { Col } from 'react-bootstrap';

const PillContainer = (props) => {
 
    return (
        <Col md={1} aria-label='Pill' value={props.pct} onClick={props.func} className='pill'><h3>{props.val}</h3></Col>
    );
};

export default PillContainer;