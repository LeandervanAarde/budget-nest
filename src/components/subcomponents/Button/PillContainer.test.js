import { render, screen } from "@testing-library/react"
import PillContainer from "./PillContainer";
import renderer from 'react-test-renderer';

const saveArr = [0.03];

describe("Testing the pill container", () =>{

    beforeEach(() =>{
        render(saveArr.map((e) => (<PillContainer pct={e} val={Math.round((e*100)) + "%"} />)));
    
    });

    test('To show the save percentages', () => {
        const element = screen.getAllByLabelText(/Pill/i) ;
        expect(element[0]).toBeInTheDocument(); 
        expect(element[0].firstChild.innerHTML).toBe("3%");
    });

    test('create a snapshot for the component', () =>{
        const element = screen.getAllByLabelText(/Pill/i) ;
        expect(element[0]).toBeInTheDocument(); 
        expect(element[0]).toMatchSnapshot();
    });
});

