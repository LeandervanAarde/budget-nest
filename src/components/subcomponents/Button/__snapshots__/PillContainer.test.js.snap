// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing the pill container create a snapshot for the component 1`] = `
<div
  aria-label="Pill"
  class="pill col-md-1"
  value="0.03"
>
  <h3>
    3%
  </h3>
</div>
`;
